### PR TITLE
homework task 1+2

### PR DESCRIPTION
**Pull request by YaroslavGitHub**
homework 2 (task 1+2) checked by eslint

### CODE REVIEW

File: homework/yarm/index.js [hunk 1]
```diff
@@ -1 +1,48 @@
-console.log(`Hello world, from Yaroslav`);
\ No newline at end of file
+<script>
+// Palindrom
+let str = '98789';
```
Comment (kovm) [inline homework/yarm/index.js]: Move all stuff below to palindrome function to encapsulate all local variables

File: homework/yarm/index.js [hunk 2]
```diff
@@ -1 +1,48 @@
...
+let textPop = ''
+let res = ''
+let i
+
+for (i = 0; i < arr.length; i++) {
+    textShift += arr.shift()
+    textPop += arr.pop()
+}
+
+function palidrom() {
```
Comment (kovm) [inline homework/yarm/index.js]: let this function gets the argument
e.f. `function palidrom(str) { ... }`
`palidrom(98789) // true`

File: homework/yarm/index.js (hunk 1 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: all stuff moved to palindrome function

File: homework/yarm/index.js (hunk 2 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: Function got its argument.
I have 1 parsing error: Unexpected token let (line 7:5) but why it happens don't understand.

File: homework/yarm/index.js [hunk 3]
```diff
@@ -1 +1,47 @@
-console.log(`Hello world, from Yaroslav`);
\ No newline at end of file
+<script>
```
Comment (kovm) [inline homework/yarm/index.js]: Remove <script> tag

File: homework/yarm/index.js [hunk 4]
```diff
@@ -1 +1,47 @@
...
+// Palindrome
+
+function palindrome(str) {
```
Comment (kovm) [inline homework/yarm/index.js]: Perform eslint check 
See my message in Slack how to do that

File: homework/yarm/index.js [hunk 5]
```diff
@@ -1 +1,47 @@
...
+        textPop += arr.pop()
+    }
+
+    if (textShift === textPop) {
+        res = 'this word is palindrome'
+    }
+    else {
+        res = 'this word is NOT palindrome'
+    }
+
```
Comment (kovm) [inline homework/yarm/index.js]: Looks like palindrome function returns nothing

File: homework/yarm/index.js [hunk 6]
```diff
@@ -1 +1,47 @@
...
+    if (textShift === textPop) {
+        res = 'this word is palindrome'
+    }
+    else {
+        res = 'this word is NOT palindrome'
+    }
+    
+}
+
+palindrome('Annaanna')
```
Comment (kovm) [inline homework/yarm/index.js]: palindrome() // ?
palindrome(null) // ?

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (kovm) [inline .vscode/settings.json]: Remove this file from commit

File: homework/yarm/index.js (hunk 3 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: Thank you, now removed

File: homework/yarm/index.js (hunk 4 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: done, executed: npx eslint homework/yarm - now no mistakes received

File: homework/yarm/index.js (hunk 5 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: fix, now return true/false

File: .vscode/settings.json (hunk 7 above)
Comment (YaroslavGitHub) [reply .vscode/settings.json]: add vscode to .gitignore and remove from repo (git rm --cashed - path...)

File: homework/yarm/index.js [hunk 8]
```diff
@@ -1 +1,36 @@
-console.log(`Hello world, from Yaroslav`);
\ No newline at end of file
+// Palindrome
+
+function palindrome (str) {
+   str = 'anna989anna';
```
Comment (kovm) [inline homework/yarm/index.js]: you rewrite input value this is a mistake your palindrome function always returns true 
I expect palindrome(33) returns true
palindrome('aaz') returns false
pay attention to invalid values e.g. palindrome(null), palindrome(), palindrome({}) etc.

Comment (YaroslavGitHub) [reply homework/yarm/index.js]: Thank you for review, requested changes done

File: homework/yarm/index.js [hunk 9]
```diff
@@ -1 +1,41 @@
-console.log(`Hello world, from Yaroslav`);
\ No newline at end of file
+// Palindrome
+
+function palindrome (str) {
```
Comment (kovm) [inline homework/yarm/index.js]: palindrome() should work with numbers too
e.g. palindrome(22)

Comment (YaroslavGitHub) [reply homework/yarm/index.js]: done

File: homework/yarm/index.js [hunk 10]
```diff
@@ -1 +1,45 @@
...
+   if (typeof str === 'number') {
+      str = str.toString();
+      str = str.toLowerCase().replace(/ /g, '');
```
Comment (kovm) [inline homework/yarm/index.js]: Take a look at \s
_Matches a white space character, including space, tab, form feed, or line feed._
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

File: homework/yarm/index.js [hunk 11]
```diff
@@ -1 +1,45 @@
-console.log(`Hello world, from Yaroslav`);
\ No newline at end of file
+// Palindrome
+
+function palindrome (str) {
+   if (typeof str === 'number') {
+      str = str.toString();
```
Comment (kovm) [inline homework/yarm/index.js]: `str = str.toString().toLowerCase().replace()`
no need on an extra line

File: homework/yarm/index.js [hunk 12]
```diff
@@ -1 +1,45 @@
...
+      str = str.toLowerCase().replace(/ /g, '');
+   }
+   if (typeof str !== 'string') {
```
Comment (kovm) [inline homework/yarm/index.js]: too many if-s and code duplication
```
if (typeof str === 'number' || typeof str === 'string') {
   str = str.toString().toLowerCase().replace(/ /g, '');
} else {
   return console.log();
}
```

File: homework/yarm/index.js [hunk 13]
```diff
@@ -1 +1,45 @@
...
+   const arr = str.split('');
+   let textShift = '';
+   let textPop = '';
+
+   for (let i = 0; i < arr.length; i++) {
+      textShift += arr.shift();
+      textPop += arr.pop();
+   }
+
+   if (textShift === textPop) {
```
Comment (kovm) [inline homework/yarm/index.js]: `return textShift === textPop;`
without any if and else

File: homework/yarm/index.js (hunk 11 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: thank you, understood & fixed

File: homework/yarm/index.js (hunk 13 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: done, remove if & else

File: homework/yarm/index.js (hunk 12 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: understood, fixed it

File: homework/yarm/index.js (hunk 10 above)
Comment (YaroslavGitHub) [reply homework/yarm/index.js]: thank you, good article, fixed my code

**Pull request merged**